core/server/models/member.js

```diff
@@ -223,6 +225,51 @@ const Member = ghostBookshelf.Model.extend({
         }
 
         return options;
+    },
+
+    async destroy(unfilteredOptions) {
```

Comment:
Kind of annoying that we've essentially duplicated the logic from the BaseModel `destroy` rather than using "super" via the prototype.

The reason for this is that we need the `id` of the model that is going to be deleted, so we can find the customers & subscriptions to delete.

`destroy` takes a `destroyBy` so it could be used to destroy by email, and we would not be passed an `id` in the options.

This means we _have_ to to the whole `forge().fetch()` dance to get the model, and at that point it felt sensible to just add the final `destroy` there too.

Reply:
What is the reason for doing this explicitly rather than using relation config and allowing bookshelf-relations to handle it?

Reply:
FWIW this doesn't need to block merge but I would like to understand.

Reply:
> What is the reason for doing this explicitly rather than using relation config and allowing bookshelf-relations to handle it?

Both Kevin and I had a call this week to plan our week and both seemed to be under the impression that the relations wouldn't work for this :thinking: Coming to this fresh this morning though that is wrong.

And for the wider question - "how come these relations weren't added originally?" - they were! And then subsequently removed again as they weren't being used and was trying to keep everything as explicit as possible.

The reason they weren't being used is because the webhook handling would/does receive information about a Subscription with no context of a Member, and just `upsert`ing the subscription information. (Rather than have the member context and then attaching the customer/subscription as children).

> this doesn't need to block merge

:+1: I think it is worth me timeboxing an hour this morning though to get this working with relations - that seems much cleaner going foward

Reply:
Ok cool yes please give it a whirl for an hour as it _would_ unblock several other things :)

Reply:
I've pushed some commits with the relations set up :relaxed: 

The initial tests still pass (yay) & I've added some more tests for the relations, it all looks good apart from one thing which I can't get working.

[This test](https://github.com/TryGhost/Ghost/pull/12077/files#diff-844c7ebbc6c104746308a04599f77a6bR16) _should_ pass, but I cannot work out why it doesn't. It means that the relations setup isn't really complete, however the functionality isn't used anywhere as of yet.

I don't want to get stuck in a rabbit hole of debugging it, especially when it isn't used/required - but I think it's not enough of a problem to warrant _not_ using the relations until it's solved, what do you think? I'd be inclined to raise it as a tech debt issue, unless you can see something immediately obvious with fresh eyes?